### PR TITLE
SALTO-4252: add workflow diagram transition initial source

### PR DESCRIPTION
**Pull request by shayc331**
_add workflow diagram transition initial source_

---
I added a new field to workflows that is called `diagramInitialEntry`

Screen Shot 2023-05-31 at 12 34 43

---
_Release Notes_: 
Jira Adapter:
* fix a bug with workflow diagram initial transition source 

---
_User Notifications_: 
Jira Adapter:
* workflows have a new field called `diagramInitialEntry` that contains the location coordinates of the initial transition source in the workflow diagram

### CODE REVIEW

**Review by alonstern — COMMENTED** [review]
Looks good 👌 We will also need noise reduction for this

File: packages/jira-adapter/src/filters/workflow/workflow_diagrams.ts [hunk 1]
```diff
@@ -275,6 +288,10 @@ const insertWorkflowDiagramFields = (workflow: WorkflowInstance,
         y: statusIdToStatus[status.id as string].y }
     }
   })
+  workflow.value.diagramInitialEntry = {
+    x: statusIdToStatus.initial.x,
```
[the remark this answers is not in the document]
Comment (shayc331) [reply packages/jira-adapter/src/filters/workflow/workflow_diagrams.ts:292]: once you create a workflow the default diagram has an initial transition that must have a source and it cannot be deleted. so I think it must be defined for any diagram. @alonstern

[the remark this answers is not in the document]
Comment (alonstern) [reply packages/jira-adapter/src/filters/workflow/workflow_diagrams.ts:292]: Still probably better to add here `?` just in case

**Review by alonstern — APPROVED** [review]
👍 👍

**Pull request merged**